test_opensearchpy/http_server.py

```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+import json
+import threading
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+hostName = "localhost"
```

Comment:
Let's avoid globals and default those inside the implementation of `HTTP_Server`.